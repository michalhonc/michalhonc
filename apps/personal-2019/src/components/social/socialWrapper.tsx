import * as React from "react";

interface IProps {
  Icon: React.ElementType | string;
  profile: string;
  profileUrl: string;
  className: string;
}

export const SocialWrapper: React.FC<IProps> = ({
  Icon,
  profile,
  profileUrl,
  children,
  className,
}) => {
  return (
    <div className="Social-wrapper">
      <div className="Social-icon">
        <Icon />
      </div>
      <a href={profileUrl} className="underline Social-profile">
        {profile}
      </a>
      <div className={"Social-content " + className}>{children}</div>
    </div>
  );
};
