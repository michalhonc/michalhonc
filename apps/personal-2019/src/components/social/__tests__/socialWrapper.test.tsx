import * as React from "react";
import renderer from "react-test-renderer";
import { SocialWrapper } from "../socialWrapper";

describe("socialWrapper", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <SocialWrapper
          Icon="0123"
          profile="@profile"
          profileUrl="https://example.com"
          className="className"
        >
          <div>Test div</div>
        </SocialWrapper>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
