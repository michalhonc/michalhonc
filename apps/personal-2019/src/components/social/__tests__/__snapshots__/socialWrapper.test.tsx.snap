// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`socialWrapper renders correctly 1`] = `
<div
  className="Social-wrapper"
>
  <div
    className="Social-icon"
  >
    <0123 />
  </div>
  <a
    className="underline Social-profile"
    href="https://example.com"
  >
    @profile
  </a>
  <div
    className="Social-content className"
  >
    <div>
      Test div
    </div>
  </div>
</div>
`;
